promql/engine_test.go

```diff
@@ -1028,3 +1029,44 @@ func TestSubquerySelector(t *testing.T) {
 		}
 	}
 }
+
+func inspectExpr(expr string, f func(p []Node) (bool, []string)) ([]bool, [][]string) {
```

Comment:
make sense, we can using `extractGroupsFromPath` directly rather than a parameter, let me also try using `TestParamsSetCorrectly ` to test grouping parameters. Thanks @bwplotka